public/components/contentSourceEdit.react.js

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import { Button, ButtonToolbar, Input, Alert } from 'react-bootstrap';
+import ContentSourceService from '../services/contentSourceService';
+
+export default class ContentSourceEdit extends React.Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            appName: this.props.appName,
+            description: this.props.description,
+            reindexEndpoint: this.props.reindexEndpoint,
+            alertStyle: 'success',
+            alertMessage: '',
+            alertVisibility: false};
+        this.exitEditMode = this.exitEditMode.bind(this);
+        this.handleAppNameChange = this.handleAppNameChange.bind(this);
+        this.handleDescriptionChange = this.handleDescriptionChange.bind(this);
+        this.handleReindexEndpointChange = this.handleReindexEndpointChange.bind(this);
+        this.handleSubmit =  this.handleSubmit.bind(this);
```

Comment:
It can be a pain to maintain all of these, the other option if they're only being used once is doing it in the onChange handler `onChange={this.handleSubmit.bind(this)}`